lib/OpenQA/WebAPI/Controller/API/V1/JobTemplate.pm

```diff
@@ -323,22 +323,23 @@ sub update {
                             my $machine_name;
                             my $settings = dclone($yaml_defaults_for_arch->{settings} // {});
                             if (ref $spec eq 'HASH') {
-                                foreach my $name (sort keys %$spec) {
-                                    my $attr = $spec->{$name};
-                                    $testsuite_name = $name;
-                                    if ($attr->{priority}) {
-                                        $prio = $attr->{priority};
-                                    }
-                                    if ($attr->{machine}) {
-                                        $machine_name = $attr->{machine};
-                                    }
-                                    if ($attr->{testsuite}) {
-                                        $job_template_name = $testsuite_name;
-                                        $testsuite_name    = $attr->{testsuite};
-                                    }
-                                    if ($attr->{settings}) {
-                                        %$settings = (%{$settings // {}}, %{$attr->{settings}});
-                                    }
+                                if (keys %$spec > 1) {
+                                    die "Only one key allowed";
```

Comment:
The message might be somewhat obscure on its own. Something like `"Invalid syntax: job template definition can't contain multiple entries"` would perhaps be more user-centric.

That said, it'd probably be even better to define `maxProperties` in the schema to catch it at that level rather than relying on the code.

Reply:
Oh, I only approved it because I was under the impression this is actually catched at a higher level and this is just an assert.

Reply:
ok, yeah, I wasn't yet sure about the error message.
I added `maxProperties` to the schema like suggested, so no manual error message required any more.
I will also add a test for the invalid case.